add support for didDeployMessage (inspired by redis plugin)

sd opened the pull request:
The ember-cli-deploy-redis plugin provides a "didDeployMessage" config option that allows it to report the revisionKey that got deployed, but not activated, along with a simple explanation about the next step required for activation.

This PR follows the same pattern for s3-index, and defaults to reporting the following after deploy:
- Deployed but did not activate revision 268206c. To activate, run: ember deploy:activate production --revision=268206c

LevelbossMike committed:
Fix quickstart instructions in README.md

`ember-cli-deploy-revision-key` is outdated. We now link to the
correct `ember-cli-deploy-revision-data`-plugin instead.

Fixes #16

sd committed:
Add support for didDeployMessage (inspired by redis plugin)

lukemelia commented:
Thanks @sd. LGTM. It would be nice to have this (and the rest of this file) under test. WDYT @LevelbossMike?

LevelbossMike commented:
LGTM as well. We should add a note about the new configuration setting in the README too.

I added some tests for index.js in https://github.com/ember-cli-deploy/ember-cli-deploy-s3-index/pull/15 We can add tests for the logging message there.

LevelbossMike commented on index.js:
```diff
@@ -26,6 +26,15 @@ module.exports = {
         },
+        didDeployMessage: function(context){
+          var revisionKey = context.revisionData && context.revisionData.revisionKey;
+          var activatedRevisionKey = context.revisionData && context.revisionData.activatedRevisionKey;
```
Where is `revisionData.activatedRevisionKey` set? In the redis case this is set in the activate hook https://github.com/ember-cli-deploy/ember-cli-deploy-redis/blob/aaea5fb1ba806335e3dfbd56f383ae7728f84549/index.js#L89 which would be easy to do in this plugin as well but seems to be missing here

sd commented:
See kids? this is what happens when you try to make a quick pull request, specially one without tests... you forget things, you embarrass yourself, and your mistakes live forever in the git log.

ghedamat commented:
@sd maybe squash in one commit after addressing @LevelbossMike comment? :p

thanks in the meantime! :)

LevelbossMike commented on index.js:
```diff
@@ -70,9 +79,24 @@ module.exports = {
 
         var s3 = new S3({ plugin: this });
-        return s3.activate(options);
+        return Promise.resolve(s3.activate(options))
```
`s3.activate` returns a promise already so no need to `Promise.resolve` here

LevelbossMike commented on index.js:
```diff
@@ -70,9 +79,24 @@ module.exports = {
+              }
+            };
+          })
+          .catch(this._errorMessage.bind(this));
```
There should already be an error reported to the console without catching here. I think `ember-cli-deploy` already takes care of this. I may be mistaken though.

sd commented:
Added info to README and squashed the commits.

Not sure why my rebase/squashing pulled in the "Fix quickstart instructions in README.md" commit too, I hope that's not a problem.

LevelbossMike commented on index.js:
```diff
@@ -90,6 +113,11 @@ module.exports = {
           });
       },
+
+      _errorMessage: function(error) {
```
We can safely remove that I guess as it isn't used anymore in the latest changes.

LevelbossMike commented:
Thx! I added a last small comment. About the squashing of the fix quickstart. it would be great if you could only submit the necessary changes in one commit without the quickstart change (maybe just redo the changes and submit a new merge request as it might get annoying to undo the rebase). As this was merged via pull request and I'm not sure if it would work otherwise without changing the history.

This is a great addition! sorry that this  takes so long to merge.

ghedamat commented:
@sd ping?

let us know if you need anything (or help with the rebase)

ghedamat commented:
@sd sorry to bug you again, would love to get this in as we approach a new release cycle for the main ember-cli-deploy project

sd commented:
I'll make some time to work on this. Sorry for the delay.

On Tue, Jan 19, 2016 at 7:58 AM Mattia Gheda notifications@github.com
wrote:

> @sd https://github.com/sd sorry to bug you again, would love to get
> this in as we approach a new release cycle for the main ember-cli-deploy
> project
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/ember-cli-deploy/ember-cli-deploy-s3-index/pull/23#issuecomment-172845492
> .

sd committed:
Merge remote-tracking branch 'upstream/master'

ghedamat commented:
no problem @sd , thanks instead!
